config/v1/types_ingress.go

```diff
@@ -41,13 +43,122 @@ type IngressSpec struct {
+type consumingUser string
+
+// hostname is an alias for hostname string validation.
+// +kubebuilder:validation:MinLength=1
+// +kubebuilder:validation:MaxLength=253
+// +kubebuilder:validation:Format=hostname
+type hostname string
+
 type IngressStatus struct {
+	// componentRoutes is where participating operators place the current route status for routes whose
+	// hostnames and serving certificates can be customized by the cluster-admin.
+	//
+	// An operator that creates entries in this slice should clean them up during removal.
+	// An operator must also handle the case of deleted status without churn.
+	// +optional
+	ComponentRoutes []ComponentRouteStatus `json:"componentRoutes,omitempty"`
 }
 
-// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// ComponentRouteSpec allows for configuration of a route's hostname and serving certificate.
+type ComponentRouteSpec struct {
+	// namespace is the namespace of the route to customize. It must be a real namespace. Using an actual namespace
+	// ensures that no two components will conflict and the same component can be installed multiple times.
+	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Required
+	// +required
+	Namespace string `json:"namespace"`
 
+	// name is the logical name of the route to customize.
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=256
+	// +kubebuilder:validation:Required
+	// +required
+	Name string `json:"name"`
+
+	// hostname is the hostname that should be supported by the route.
+	// +kubebuilder:validation:Required
+	// +required
+	Hostname hostname `json:"hostname"`
+
+	// servingCertKeyPairSecret is a reference to a secret in namespace/openshift-config that is a kubernetes tls secret.
+	// The serving cert/key pair must match and will be used by the operator to fulfill the intent of serving with this name.
+	// +optional
+	ServingCertKeyPairSecret SecretNameReference `json:"servingCertKeyPairSecret"`
+}
+
+// ComponentRouteStatus contains information allowing configuration of a route's hostname and serving certificate.
+type ComponentRouteStatus struct {
+	// namespace is the namespace of the route to customize. It must be a real namespace. Using an actual namespace
+	// ensures that no two components will conflict and the same component can be installed multiple times.
+	// +kubebuilder:validation:Pattern=^[a-z0-9]([-a-z0-9]*[a-z0-9])?$
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=63
+	// +kubebuilder:validation:Required
+	// +required
+	Namespace string `json:"namespace"`
+
+	// name is the logical name of the route to customize. It does not have to be the actual name of a route resource.
+	// Keep in mind that this is your API for users to customize. You could later rename the route, but you cannot rename
+	// this name.
```

Comment:
Who is the audience?  Maybe s/your/the/, /You could/An operator may/, and s/you/it/.